external-crates/move/move-compiler/src/sui_mode/typing.rs

```diff
@@ -403,11 +411,12 @@ fn init_signature(context: &mut Context, name: FunctionName, signature: &Functio
     } else if parameters.len() > 1 {
         // if there is more than one parameter, the first must be the OTW
         let (_, first_var, first_ty) = parameters.first().unwrap();
-        let is_otw = matches!(
-            first_ty.value.type_name(),
-            Some(sp!(_, TypeName_::ModuleType(m, n)))
-                if m == context.current_module() && n.value() == otw_name
-        );
+        let is_otw = matches!(&first_ty.value, Type_::UnresolvedError | Type_::Var(_))
```

Comment:
This is the only line that I don't really follow. Is this meant to allow for good error recovery for something like the following two?

```
module a::boop {
   fun init(_: BOOP, _: &mut sui::tx_context::TxContext) { ... }
}
```
and 

```
module a::boop {
   fun init<BOOP>(_: BOOP, _: &mut sui::tx_context::TxContext) { ... }
}
```

If so it may be worthwhile adding tests for these cases.

Reply:
I added tests for at least the first one. `Type::Var(_)` would be an unexpanded type variable. 

This is really trying to mimic what would happen if this was done during type checking instead... and maybe I should move some of that logic there